lib/ansible/modules/cloud/azure/azure_rm_manageddisk.py

```diff
@@ -302,7 +317,7 @@ def exec_module(self, **kwargs):
         if self.managed_by or self.managed_by == '':
             vm_name = parse_resource_id(disk_instance.get('managed_by', '')).get('name') if disk_instance else None
             vm_name = vm_name or ''
-            if self.managed_by != vm_name:
+            if self.managed_by != vm_name or self.is_attach_caching_option_different(vm_name, result):
```

Comment:
What if the vm_name is empty?

Reply:
> What if the vm_name is empty?

Thank you very much yuwei! It's a bug, I fixed it and modify the documentation.